ledger/src/lib.rs

```diff
@@ -105,6 +109,8 @@ pub struct Ledger<N: Network, C: ConsensusStorage<N>> {
     current_committee: Arc<RwLock<Option<Committee<N>>>>,
     /// The current block.
     current_block: Arc<RwLock<Block<N>>>,
+    /// A cache containing the list of recent succesfully checked transactions.
+    checked_transactions: Arc<RwLock<LruCache<N::TransactionID, Transaction<N>>>>,
```

Comment:
We may want to purge the cached transaction once they are included in a block.

Reply:
We also likely don't need to cache verification of execution transactions. Because it's a simple proof check (~20ms), which is likely not worth storing another tx in memory and taking up a cache spot. 

In addition, if a node receives 1 deployment and then 128 executions, the deployment verification (which is the important one) gets purged from the cache.

Reply:
I think it is safe to only store the transaction_id in the cache. It should be deterministic from the Transaction. We do not check this when receiving the Transmission, but it is the second check inside `check_transaction`.

As for whether this is desired, let's assume the cache grows as big as five blocks worth of transactions. In theory blocks can have at most (`MAX_TRANSACTIONS`) 65536 transactions. In practice this might be up to 600 executions, or maybe five large deployments. So we're looking at a size of between 5*600 and 5*65536 transactions. If we only store IDs (32 bytes) this is between 100KB and 10MB. If we store the transactions (e.g. 1.5 KB execution), that would be between 4MB and 491 MB. 

4MB actually seems pretty doable, so if we get a big performance gain and this is a "hot path", we could choose to store transactions.

I would however advocate for also storing Executions, but this can also be in a future PR.

Reply:
Yes, based on a recent Slack discussion, my understanding is that the ID of a transaction is cryptographically calculated as part of deserializing the transaction (https://github.com/AleoHQ/snarkVM/blob/0f324520e22d9af9d855be41893355f41a1efc63/ledger/block/src/transaction/bytes.rs#L63), as soon as the transaction enters snarkOS (e.g. via a message). In other words, it should be an invariant of the transaction data structure that the ID is cryptographically tied to the rest of the transaction, and therefore that transaction IDs are (practically) unique. This is a critical property to justify the handling of transactions by their IDs in various parts of snarkOS.